contracts/script/EnvironmentParser.s.sol

```diff
@@ -0,0 +1,130 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.9;
+
+import {Script} from "forge-std/Script.sol";
+import {console} from "forge-std/console.sol";
+import {Vm} from "forge-std/Vm.sol";
+import {StdStyle} from "forge-std/StdStyle.sol";
+
+interface IEigenDADirectory {
+    function getAllNames() external view returns (string[] memory);
+    function getAddress(string memory name) external view returns (address);
+}
+
+interface IOwnable {
+    function owner() external view returns (address);
+}
+
+/// Warning: Assumes the directory correctly contains all contracts for a given network.
+
+contract EnvironmentParser is Script {
+    // ERC1967 storage slots
+    bytes32 constant IMPLEMENTATION_SLOT = 0x360894a13ba1a3210667c828492db98dca3e2076cc3735a920a3ca505d382bbc;
+    bytes32 constant ADMIN_SLOT = 0xb53127684a568b3173ae13b9f8a6016e243e63b6e8ee1178d6a717850b5d6103;
+
+    struct ContractInfo {
+        string name;
+        address addr;
+        bool isProxy;
+        bool isOwnable;
+        address implementation;
+        address admin;
+        address owner;
+    }
+
+    function run(address directoryAddress) external {
+        console.log(StdStyle.bold(StdStyle.blue("=== EigenDA Registry Audit ===")));
+        console.log(StdStyle.cyan("Directory:"), directoryAddress);
+
+        IEigenDADirectory directory = IEigenDADirectory(directoryAddress);
+
+        // Step 1: Get all names
+        string[] memory names = directory.getAllNames();
+        console.log(StdStyle.green("Found"), StdStyle.bold(names.length), StdStyle.green("registered contracts"));
+        console.log("");
+
+        // Step 2-5: Iterate and audit each
+        ContractInfo[] memory results = new ContractInfo[](names.length);
+
+        for (uint256 i = 0; i < names.length; i++) {
+            console.log(StdStyle.magenta(string.concat("[", vm.toString(i + 1), "/", vm.toString(names.length), "]")), StdStyle.bold(names[i]));
+
+            ContractInfo memory info;
+
+            info.name = names[i];
+
+            // Step 2: Get address for this name
+            info.addr = directory.getAddress(names[i]);
+
+            // Step 4: Check ERC1967 slots (tells us if it's a proxy)
+            info.implementation = address(uint160(uint256(vm.load(info.addr, IMPLEMENTATION_SLOT))));
+            info.admin = address(uint160(uint256(vm.load(info.addr, ADMIN_SLOT))));
+            info.isProxy = info.implementation != address(0);
+
+            // Step 5: Query owner
```

Comment:
some other access management to be aware of:

**Registry Coordinator**
- [_pauser_](https://github.com/Layr-Labs/eigenda/blob/132fbbaa0d2e52744fee104165173dda0150599e/contracts/src/core/EigenDARegistryCoordinator.sol#L215-L217) that can halt operator set modification
- [_ejector_](https://github.com/Layr-Labs/eigenda/blob/132fbbaa0d2e52744fee104165173dda0150599e/contracts/src/core/EigenDARegistryCoordinator.sol#L286-L307) that can eject existing operators